packages/sdk-js/src/GrowthBook.ts

```diff
@@ -733,13 +736,22 @@ export class GrowthBook<
         if (navigate) {
           if (isBrowser) {
             this._setAntiFlicker();
-            window.setTimeout(() => {
+            // Wait for the possibly-async tracking callback, bound by min and max delays
+            Promise.all([
+              promiseTimeout(
+                this._activeTrackingCall,
+                this._ctx.maxNavigateDelay ?? 10000
+              ),
+              new Promise((resolve) =>
+                window.setTimeout(resolve, this._ctx.navigateDelay ?? 100)
+              ),
+            ]).then(() => {
               try {
                 navigate(url);
               } catch (e) {
                 console.error(e);
               }
-            }, this._ctx.navigateDelay ?? 100);
+            });
           } else {
             try {
               navigate(url);
```

Comment:
wondering if we should also do something similar for non-browser contexts. I do not think we'd need this on edge fwiw.